app/controllers/api/v7/works_controller.rb

```diff
@@ -138,6 +158,14 @@ def get_ids(params)
       collection = Work.where(publisher_id: publisher.id)
     elsif params[:contributor_id] && contributor = Contributor.where(pid: params[:contributor_id]).first
       collection = Work.joins(:contributions).where("contributions.contributor_id = ?", contributor.id)
+    elsif params[:id]
+      id_hash = get_id_hash(params[:id])
+      if id_hash.respond_to?("key")
```

Comment:
Yes, I guessed it was something like this. Could you just put a comment in to explain the two types of input you expect? Without is unmaintainable and confusing.